spec/ruby/core/kernel/yield_self_spec.rb

```diff
@@ -0,0 +1,21 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/classes', __FILE__)
+
+describe "Kernel#yield_self" do
+  it "always yields self" do
+    a = KernelSpecs::A.new
+    a.yield_self { |o| o.should equal a }
+  end
+
+  it "always returns the block return value" do
```

Comment:
Minor, but since we specify the entire behavior, I think the `always` is redundant.